pylabnet/scripts/lasers/laser_stabilizer.py

```diff
@@ -0,0 +1,304 @@
+from pylabnet.scripts.pid import PID
+from pylabnet.network.core.service_base import ServiceBase
+from pylabnet.network.core.client_base import ClientBase
+from pylabnet.gui.pyqt.external_gui import Window
+from pylabnet.utils.helper_methods import (unpack_launcher, create_server,
+    load_config, get_gui_widgets, get_legend_from_graphics_view, add_to_legend)
+from pylabnet.utils.logging.logger import LogClient, LogHandler
+import pylabnet.hardware.ni_daqs.nidaqmx_card as nidaqmx
+import pylabnet.hardware.staticline.staticline as staticline
+import pylabnet.network.client_server.nidaqmx_card as nidaqmx_card_server
+
+import numpy as np
+import time
+import copy
+import pickle
+import socket
+import pyqtgraph as pg
+
+
+class LaserStabilizer:
+    """A class for stabilizing the laser power given a DAQ input, a power control output, and a setpoint"""
+    def __init__(self, config='toptica_laser_stabilization'):
+        """Instantiates LaserStabilizer script object for stabilizing the laser
+             :param config: (str) name of config file """
+                
+        # Instantiate GUI
+        self.gui = Window(
+            gui_template='power_stabilizer',
+            host=socket.gethostbyname(socket.gethostname())
+        )
+        self.widgets = get_gui_widgets(self.gui, p_setpoint=1, p_outputVoltage=1, label_power=1, config=1,
+            graph=2, legend=2, clients=1, exp=1, exp_preview=1, configure=1, start=1,
+            autosave=1, save_name=1, stop=1, clear=1)
+
+        self.widgets['config'].setText(config)
+        self._load_config_file(config)
+        
+        #Now initialize control/output voltage to 0, and set up label
+        self._curr_output_voltage = self.widgets['p_outputVoltage'].value() #Stores current output voltage that is outputted by the AO
+        self.widgets['p_outputVoltage'].valueChanged.connect(self._set_output_voltage_from_label)
+        self._ao_client.set_ao_voltage(self._ao_channel, self._curr_output_voltage)
+
+        #Update the input power label
+        self._last_power_text_update = 0
+        self._update_power_label()
+
+        self._initialize_graphs()
+
+        #Finally hookup the final buttons
+        self.widgets['start'].clicked.connect(lambda: self.start(update_st_gui=True))
+        self.widgets['stop'].clicked.connect(self.stop)
+        self.widgets['clear'].clicked.connect(lambda: self._clear_data_plots(display_pts=5000))
+
+        #Initially the program starts in the "unlocked" phase
+        self._is_stabilizing = False
+
+    def _load_config_file(self, config):
+        """Loads the config file"""
+        #Now load the config file 
+        self.config = load_config(config, logger=None)
```

Comment:
let's use the launcher technique to pass port info/ instantiate clients rather than config (for now). We can discuss this, and in the meantime you can see `pylabnet.launchers` and `launch(**kwargs)` functions in various other script modules to see how this is done